oclubs/blueprints/resblueprint.py

```diff
@@ -0,0 +1,461 @@
+    timeslot = ActivityTime[request.form.get('timeslot').upper()]
+    single_date = int_to_dateobj(
+        int(request.form.get('date_selection').replace('-', '')))
+
+    rdict = Classroom.get_free_classroom_conditions(
+        buildings=building,
+        timeslot=timeslot,
+        dates=single_date)
+
+    free_classrooms = rdict[building][timeslot][single_date]
+
+    if free_classrooms:
+        choices = [(r.room_number, r.room_number) for r in free_classrooms]
+    else:
+        choices = [('none', 'None available')]
+
+    # update the choices on the client side
+    return jsonify(choices)
+
+
+@resblueprint.route('viewres/club/<club>',  methods=['GET', 'POST'])
+@get_callsign_decorator(Club, 'club')
+def viewreservations_club(club):
+    is_admin = False
+    is_owner = False
+
+    if current_user.is_authenticated:
+        if current_user.type == UserType.CLASSROOM_ADMIN or \
+                current_user.type == UserType.DIRECTOR:
+            is_admin = True
+
+        if current_user == club.leader:
+            is_owner = True
+
+    res = Reservation.get_reservations_conditions(
+        reserver_club=club)
+
+    canReserveForm = ChangeCanReservationForm(
+        changeCanReserve=str(club.reservation_allowed))
+    canSBForm = ChangeCanSmartboardForm(
+        changeCanSB=str(club.smartboard_allowed))
+
+    if request.method == 'POST':
+        if canReserveForm.submit.data:
+            club.reservation_allowed = (
+                True
+                if canReserveForm.changeCanReserve.data == 'True'
+                else False)
+
+        if canSBForm.submit.data:
+            club.smartboard_allowed = (
+                True
+                if canSBForm.changeCanSB.data == 'True'
+                else False)
+
+        return redirect(url_for('.viewreservations_club', club=club.callsign))
+
+    return render_template('reservation/viewres_club.html.j2',
+                           res=res,
+                           is_admin=is_admin,
+                           is_owner=is_owner,
+                           canReserveForm=canReserveForm,
+                           canSBForm=canSBForm)
+
+
+@resblueprint.route('<reservation>/delete')
+@get_callsign_decorator(Reservation, 'reservation')
+def deletereservation(reservation):
+    club = reservation.reserver_club
+    single_date = reservation.date
+    timeslot = reservation.timeslot
+    building = reservation.classroom.building
+    room_number = reservation.classroom.room_number
+    owner = current_user
+
+    if reservation.status == 1:
+        pass
+    elif reservation.status == 2:
+        reservation.activity.reservation = None
+
+    try:
+        ret = Reservation.delete_reservation(single_date,
+                                             timeslot,
+                                             building,
+                                             room_number,
+                                             owner)
+    except NoRow:
+        abort(500)
```

Comment:
Don't abort(500). that doesn't explain the issue with a Traceback. NoRow will generate a 500 anyways